engine/src/main/scala/cromwell/engine/backend/sge/SgeBackend.scala

```diff
@@ -36,6 +36,9 @@ object SgeBackend {
     def script = jobDescriptor.callRootPath.resolve("script.sh")
     def returnCode = jobDescriptor.callRootPath.resolve("rc")
   }
+
+  private lazy val oldGoogleConf = GoogleConfigurationAdapter.gcloudConf.get
+  lazy val gcsConf = Try(GoogleConfiguration(oldGoogleConf.appName, oldGoogleConf.userAuthMode.getOrElse(oldGoogleConf.cromwellAuthMode)))
```

Comment:
Oh I think I misunderstood what @cjllanwarne was saying. :frowning: 

Over on my branch, for the specific case of GcsFilesystemSupport I've pulled out a trait that both SharedFilesystemBackend and JesBackend mix in to keep things DRY.  I was thinking this could be just a utility trait that hangs out in the backend subproject and backend impls could mix it in if they want to support GCS.  Would that address your concern without introducing any structural badness?

```scala
trait CanUseGcsFilesystem { self: Backend =>

  protected def gcsFilesystem(options: WorkflowOptions): Option[FileSystem] = {

    def gcsFilesystemMustHaveAuth: Option[String] = {
      backendConfig.getConfigOption("filesystems.gcs") map {
        _.getStringOr("auth", throw new RuntimeException(s"No auth in filesystems.gcs for backend '$name.")) }
    }

    def gcsFilesystemAuthMustBeLegit(authName: String): GoogleAuthMode = {
      def prependName(errs: NonEmptyList[String]): String = s"For backend '$name': " + errs.list.mkString("\n")

      GoogleConfiguration.Instance.auth(authName) leftMap prependName valueOr { msg => throw new RuntimeException(msg) }
    }

    val gcsAuth: Option[GoogleAuthMode] = for {
      // Throw if there's a filesystems.gcs but no auth within that.
      authName <- gcsFilesystemMustHaveAuth
      // Throw if there's an auth named in this backend's `filesystems.gcs` which isn't in the global `google` list of auths.
      auth = gcsFilesystemAuthMustBeLegit(authName)
    } yield auth

    val maybeStorage = gcsAuth map { _.buildStorage(options) }
    maybeStorage map GcsFileSystemProvider.apply map { _.getFileSystem }
  }
}
```